docs/user-guide/migrating-to-6.0.0.md

```diff
@@ -209,14 +266,30 @@ If you're not sure which config file needs to be updated, it may be useful to ru
 
 **Related issue(s):** [eslint/eslint#9990](https://github.com/eslint/eslint/issues/9990)
 
+## <a name="unicode-regexes"></a> User-provided regular expressions in rule options are parsed with the unicode flag
+
+Rules like [`max-len`](/docs/rules/max-len), accept a string option which is interpreted as a regular expression. In ESLint v6.0.0, these regular expressions are interpreted with the [unicode flag](https://mathiasbynens.be/notes/es6-unicode-regex), which should exhibit more reasonable behavior when matching characters like astral symbols. Unicode regexes also validate escape sequences more strictly than non-unicode regexes.
```

Comment:
Nit: Comma after link to "max-len" could be removed. (On mobile so can't use suggestion, sorry)